custom_components/tuya_local/devices/zemismart_spm02v25_energymeter.yaml

```diff
@@ -0,0 +1,177 @@
+name: Energy meter
+products:
+  - id: iwn0gpzz
+    manufacturer: Zemismart
+    model: "SPM02 Zigbee V2.5"
+entities:
+  - entity: sensor
+    class: energy
+    translation_key: energy_consumed
+    dps:
+      - id: 1
+        type: integer
+        name: sensor
+        class: total_increasing
+        unit: kWh
+        mapping:
+          - scale: 100
+
+  - entity: binary_sensor
+    class: problem
+    category: diagnostic
+    dps:
+      - id: 9
+        type: bitfield
+        name: sensor
+        mapping:
+          - dps_val: 0
+            value: false
+          - value: true
+      - id: 9
+        type: bitfield
+        name: fault_code
+      - id: 9
+        type: bitfield
+        name: sensor
+        mapping:
+          - mask: 0x1
+            value: Short Circuit
+          - mask: 0x2
+            value: Surge
+          - mask: 0x4
+            value: Overload
+          - mask: 0x8
+            value: Leakage Current
+          - mask: 0x10
+            value: Temp Diff Fault
+          - mask: 0x20
+            value: Fire Alarm
+          - mask: 0x40
+            value: High Power
+          - mask: 0x80
+            value: Self Test Fail
+          - mask: 0x100
+            value: Over Current
+          - mask: 0x200
+            value: Unbalance
+          - mask: 0x400
+            value: Over Voltage
+          - mask: 0x800
+            value: Under Voltage
+          - mask: 0x1000
+            value: Missing Phase
+          - mask: 0x2000
+            value: Power Outage
+          - mask: 0x4000
+            value: Magnetic Interference
+          - mask: 0x8000
+            value: Credit Alarm
+          - mask: 0x10000
+            value: No Balance
+
+  - entity: button
+    class: restart
+    category: config
+    dps:
+      - id: 12
+        type: boolean
+        name: button
+
+  - entity: text
+    category: diagnostic
+    hidden: true
+    dps:
+      - id: 17
+        type: string
+        name: sensor
+        optional: true
+
+  - entity: text
+    category: diagnostic
+    hidden: true
+    dps:
+      - id: 18
+        type: string
+        name: sensor
+        optional: true
+
+  - entity: sensor
+    class: energy
+    translation_key: energy_produced
+    state_class: sensor
+    dps:
+      - id: 23
+        type: integer
+        name: sensor
+        unit: kWh
+        mapping:
+          - scale: 100
+
+  - entity: sensor
+    class: frequency
+    state_class: measurement
+    dps:
+      - id: 32
+        type: integer
+        name: sensor
+        unit: Hz
+        mapping:
+          - scale: 100
+
+  - entity: binary_sensor
```

Comment:
In many devices, this seems to be not a sensor, but can be used to force the device to send real-time updates for the power, current, voltage readings for some time period.

If that is the case here, it would be best to make it a button that can be pressed manually or by an automation as needed.

```yaml
- entity: button
  name: Realtime updates
  dps:
    - id: 35
      type: string
      optional: true
      name: button
      mapping:
        - dps_val: online
          value: true
```